### DISCUSSION TITLE
Latest Battery Charge limit implementation.

### DISCUSSION THREAD

**Discussion by Constrat** [seerge/g-helper#3098] [Q&A] [answered]
I noticed that with https://github.com/seerge/g-helper/releases/tag/v0.189, https://github.com/seerge/g-helper/commit/421dc0c05c7370825f0052c4ac6463596edbeb19 implemented a battery limit that should be applied a bit earlier compared to having to wait for the user logon. 

The question might be stupid, but after checking the code and the correct creation of the task I notice that GHelper.exe is run with the `charge` parameter, which definitely makes sense, but my question is: can we just run GHelper.exe at System Startup? Instead of running the battery limit and then exiting the application and then rerunning the application on user logon?
Or the user logon is necessary because %appdata% is somehow locked leading to GHelper missing the necessary config.json?
I tried the "quick" battery limit and the 60% limit was correctly set, meaning before the user logon the config was accesible.

TL;DR Instead of just task scheduling on system startup `GHelper.exe charge`, the possibility of just changing the normal startup task on System Startup (without manually editing the task scheduler obvisouly)
![image](https://github.com/user-attachments/assets/b79a7db6-f446-4657-b120-52e3b9c084e9)

https://github.com/seerge/g-helper/blob/0a89062b52bcfa98bcaf313f4e9c61e1f7e8603c/app/Program.cs#L56-L61

This is mainly a question, not really a feature request. The apparent feature request is just to prove an example of a plausible usage.

As always, thank you for pouring your time and effort into this project. Your dedication is greatly appreciated!

**Comment by seerge** [comment, marked as answer]
@Constrat if you run "main" GHelper.exe process on system startup it just won't show icon in system tray and windows refuses to create main app window. So UI becomes completely inaccessible. (you can verify it by yourself)

It's either "service-style" process on startup, or a regular UI-app on logon.

I don't want to create real service, clone different processes, etc - so G-Helper just schedues it's own exe with a special param to do only thing that is worth running before login - charge limit :)

**Reply by Constrat** [reply]
I was expecting a windows limitation on why it was impossible to do so. Thank you very much.

> (you can verify it by yourself)

I indeed tried this myself around 3 months ago, to see if I could speed up the battery limit setup, but without any positive outcomes :P
